tests/e2e/e2e_util_test.go

```diff
@@ -0,0 +1,185 @@
+		err,
+		"failed connect chains; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
+	)
+
+	s.Require().Containsf(
+		errBuf.String(),
+		"successfully opened init channel",
+		"failed to connect chains via IBC: %s", errBuf.String(),
+	)
+
+	s.T().Logf("connected %s and %s chains via IBC", s.chainA.id, s.chainB.id)
+}
+
+func (s *IntegrationTestSuite) sendIBC(srcChainID, dstChainID, recipient string, token sdk.Coin) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("sending %s from %s to %s (%s)", token, srcChainID, dstChainID, recipient)
+
+	exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
+		Context:      ctx,
+		AttachStdout: true,
+		AttachStderr: true,
+		Container:    s.hermesResource.Container.ID,
+		User:         "root",
+		Cmd: []string{
+			"hermes",
+			"tx",
+			"raw",
+			"ft-transfer",
+			dstChainID,
+			srcChainID,
+			"transfer",  // source chain port ID
+			"channel-0", // since only one connection/channel exists, assume 0
+			token.Amount.String(),
+			fmt.Sprintf("--denom=%s", token.Denom),
+			fmt.Sprintf("--receiver=%s", recipient),
+			"--timeout-height-offset=1000",
+		},
+	})
+	s.Require().NoError(err)
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+		Context:      ctx,
+		Detach:       false,
+		OutputStream: &outBuf,
+		ErrorStream:  &errBuf,
+	})
+	s.Require().NoErrorf(
+		err,
+		"failed to send IBC tokens; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
+	)
+
+	s.T().Log("successfully sent IBC tokens")
+}
+
+func queryOsmoTx(endpoint, txHash string) error {
+	resp, err := http.Get(fmt.Sprintf("%s/cosmos/tx/v1beta1/txs/%s", endpoint, txHash))
+	if err != nil {
+		return fmt.Errorf("failed to execute HTTP request: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("tx query returned non-200 status: %d", resp.StatusCode)
+	}
+
+	var result map[string]interface{}
+	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+		return fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	txResp := result["tx_response"].(map[string]interface{})
+	if v := txResp["code"]; v.(float64) != 0 {
+		return fmt.Errorf("tx %s failed with status code %v", txHash, v)
+	}
+
+	return nil
+}
+
+func queryOsmoAllBalances(endpoint, addr string) (sdk.Coins, error) {
+	resp, err := http.Get(fmt.Sprintf("%s/cosmos/bank/v1beta1/balances/%s", endpoint, addr))
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute HTTP request: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	bz, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var balancesResp banktypes.QueryAllBalancesResponse
+	if err := cdc.UnmarshalJSON(bz, &balancesResp); err != nil {
+		return nil, err
+	}
+
+	return balancesResp.Balances, nil
+}
+
+func queryOsmoDenomBalance(endpoint, addr, denom string) (sdk.Coin, error) {
```

Comment:
Seems to be unused?

Reply:
Also, from what I remember this wouldn't work. I was adapting `TestQueryBalances` from this test, and I think it wouldn't work in osmosis if we were to run it

Reply:
agreed, removed